algo-core/ortools_solver.py

```diff
@@ -242,483 +308,1054 @@ def print_final_schedules(schedule_results):
+            for h in agents_vet:
+                if not (h in unavailable_agents[d]):
+                    model.AddBoolOr(v_tdh.loc[(t, d, h), "is_in_pref_range"])
 
                     for (j, sec) in enumerate(
-                        df_agents.loc[h, "HourRanges"][d]
+                        df_agents.loc[h, "hour_ranges"][d]
                     ):
+
                         model.Add(
-                            v_tdh.loc[(t, d, h), "Start"] >= sec[0]
+                            v_tdh.loc[(t, d, h), "shift_start"] >= sec[0]
                         ).OnlyEnforceIf(
-                            v_tdh.loc[(t, d, h), "IsInPrefRange"][j]
+                            v_tdh.loc[(t, d, h), "is_in_pref_range"][j]
                         )
                         model.Add(
-                            v_tdh.loc[(t, d, h), "Start"]
-                            + v_tdh.loc[(t, d, h), "Duration"]
+                            v_tdh.loc[(t, d, h), "shift_start"]
+                            + v_tdh.loc[(t, d, h), "shift_duration"]
                             <= sec[1]
                         ).OnlyEnforceIf(
-                            v_tdh.loc[(t, d, h), "IsInPrefRange"][j]
+                            v_tdh.loc[(t, d, h), "is_in_pref_range"][j]
                         )
 
-    # Constraint: Ensure agent not scheduled for more than one track at a time:
+
+def constraint_avoid_assigning_agent_multiple_tracks_per_day():
+    """Ensure each veteran is scheduled in at most 1 track per day."""
     for d in range(num_days):
-        for h in df_agents.index:
+        for h in agents_vet:
+            is_agent_on_list = []
+
+            if d == 0:  # Applies to extra Monday track as well.
+                max_tracks = num_tracks + 1
+            else:
+                max_tracks = num_tracks
+
+            for t in range(max_tracks):
+                is_agent_on_list.append(v_tdh.loc[(t, d, h), "is_agent_on"])
+
+            model.Add(sum(is_agent_on_list) <= 1)
+
+
+def constraint_honour_agent_availability_onboarding():
+    """Make sure that each onboarder's availability is honoured.
+
+    Each shift must start and end within that agent's available hours.
+    """
+    for d in range(num_days):
+        for h in agents_onb:
+            if not (h in unavailable_agents[d]):
+                model.AddBoolOr(v_dh_on.loc[(d, h), "is_in_pref_range"])
+
+                for (j, sec) in enumerate(df_agents.loc[h, "hour_ranges"][d]):
+
+                    model.Add(
+                        v_dh_on.loc[(d, h), "shift_start"] >= sec[0]
+                    ).OnlyEnforceIf(v_dh_on.loc[(d, h), "is_in_pref_range"][j])
+                    model.Add(
+                        v_dh_on.loc[(d, h), "shift_start"]
+                        + v_dh_on.loc[(d, h), "shift_duration"]
+                        <= sec[1]
+                    ).OnlyEnforceIf(v_dh_on.loc[(d, h), "is_in_pref_range"][j])
+
+
+def constraint_various_custom_conditions():
```

Comment:
I suggest encoding such custom conditions within the input file instead of putting them in the solver script. Especially because this is a public repo, it's more problematic. Currently we pass the responsibility to the (non balena team member) readers to patch the script in order to remove such custom conditions.

Reply:
I agree :+1:

Reply:
@juanfridano could you please refactor this function as well before merging this PR? 

This function has hardcoded references to balena team members. There are couple of other references to `@georgiats` and `@wrboyce` as well elsewhere in the code. Since this is a public repo, I suggest removing all such references. A user outside of balena with different settings (thus not having these team members) will not be able to run the code.